lib/Driver/Job.cpp

```diff
@@ -403,6 +407,30 @@ void Job::printSummary(raw_ostream &os) const {
   os << "}";
 }
 
+static void
+writeResponseFile(llvm::raw_ostream &OS, const llvm::opt::ArgStringList &Arguments) {
+  // wrap arguments in quotes to ensure compatibility with Unix and Windows
+  for (const char *arg : Arguments) {
+    OS << '"';
+    for (; *arg != '\0'; arg++) {
+      if (*arg == '\"' || *arg == '\\') {
+        OS << '\\'; // escape quote and backslash characters
+      }
+      OS << *arg;
+    }
+    OS << "\" ";
+  }
+}
+
+void Job::writeArgsToResponseFile() const {
+  std::string responseContents;
+  llvm::raw_string_ostream SS(responseContents);
```

Comment:
You can use `raw_fd_ostream` here and just write directly to the file. No need to write to an in-memory buffer and then export it.

Also, you should probably have a way of handling failure here. (What if the temporary directory isn't writable?)

Reply:
I'm not sure of the right way to handle a potential error since I don't appear to have access to the Diagnostic Engine within `Job`. There is an example in [`Compilation.cpp`](https://github.com/apple/swift/blob/9cc534a05a94bacb7162d361cc803aa4f9fc0f4d/lib/Driver/Compilation.cpp#L1009) where `raw_fd_ostream` is used, but there is a `FIXME` for the error handling.

Reply:
It could just returning a bool or an llvm::Error from the function for now.

Reply:
(Note that `writeFileWithEncoding` can fail just as much as `raw_fd_ostream` can.)